src/wrappers/dashboards.ts

```diff
@@ -163,27 +234,171 @@ export default class {
     await this.cloneLabels(original, createdDashboard);
 
     return this.get(createdDashboard.id);
+  }
+
+  public async createFromTemplate(
+    template: IDashboardTemplate,
+    orgID: string,
+  ): Promise<IDashboard> {
+    const { content } = template;
+
+    if (
+      content.data.type !== TemplateType.Dashboard ||
+      template.meta.version !== "1"
+    ) {
+      throw new Error("Can not create dashboard from this template");
+    }
+
+    const { name, description } = content.data.attributes;
+
+    const createdDashboard = await this.create({ orgID, name, description });
+
+    await this.createIncludedLabelsFromTemplate(template, createdDashboard);
+    await this.createIncludedCellsFromTemplate(template, createdDashboard);
+
+    const dashboard = await this.get(createdDashboard.id);
+
+    return addDefaults(dashboard);
+  }
+
+  private async createIncludedLabelsFromTemplate(
+    template: IDashboardTemplate,
+    dashboard: IDashboard,
+  ) {
+    const { content } = template;
+
+    if (
+      !content.data.relationships ||
+      !content.data.relationships[TemplateType.Label]
+    ) {
+      return;
+    }
+
+    const labelRelationships =
+      content.data.relationships[TemplateType.Label].data;
+
+    const includedResources = content.included || [];
+
+    const labelsToCreate = includedResources.reduce((acc, ir) => {
+      if (ir.type === TemplateType.Label) {
+        const found = labelRelationships.some((lr) => lr.type === TemplateType.Label && lr.id === ir.id);
+        if (found) {
+          acc = [...acc, ir];
+        }
+      }
+      return acc;
+    }, [] as ILabelIncluded[]);
+
+    const pendingLabels = labelsToCreate.map((l) => {
+      const { attributes: { name, properties } } = l;
+      return this.labelsService.labelsPost({ name, properties });
+    });
+
+    const labelsResponse = await Promise.all(pendingLabels);
+
+    const createdLabels = labelsResponse
+      .map((lr) => lr.data.label)
+      .filter((cl): cl is Label => !!cl)
+      .map((l) => l.id)
+      .filter((id): id is string => !!id);
+
+    if (!dashboard || !dashboard.id) {
+      throw new Error("Can not add labels to undefined Dashboard");
+    }
+
+    await this.addLabels(dashboard.id, createdLabels);
+  }
+
+  private async createIncludedCellsFromTemplate(template: IDashboardTemplate,
+                                                createdDashboard: IDashboard) {
+
+    const { content } = template;
+
+    if (
+      !content.data.relationships ||
+      !content.data.relationships[TemplateType.Cell]
+    ) {
+      return;
+    }
+
+    const cellRelationships =
+      content.data.relationships[TemplateType.Cell].data;
+
+    const includedResources = content.included || [];
+
+    const cellsToCreate = includedResources.reduce((acc, ir) => {
```

Comment:
Lets rename this functions, to createCellsFromTemplate, consumers should not need to think about json api details and lets filter resources and create a private method to create cells from that filtered list. This function is too complex.